src/gui/dialogs/editor/edit_unit.cpp

```diff
@@ -849,15 +841,29 @@ void editor_edit_unit::load_movetype() {
 	}
 }
 
-void editor_edit_unit::write_macro(std::ostream& out, int level, std::string macro_name)
+void editor_edit_unit::write_macro(std::ostream& out, unsigned level, const std::string macro_name)
 {
-	for(int i = 0; i < level; i++)
+	for(unsigned i = 0; i < level; i++)
 	{
 		out << "\t";
 	}
 	out << "{" << macro_name << "}\n";
 }
 
+void editor_edit_unit::write_key_val(std::ostream& out, unsigned level, const std::string& key,	const config::attribute_value& value, const bool translatable)
+{
+	for(unsigned i = 0; i < level; i++)
+	{
+		out << "\t";
+	}
+
+	if (translatable) {
+		out << key << "=" << "_ \"" << value << "\"\n";
+	} else {
+		out << key << "=" << value << "\n";
+	}
```

Comment:
What are the chances that people put a quote in their strings? This will break if they do. The non-translatable way will also break if they put curly braces or plus signs, but that seems like less of an issue.

Reply:
That method turned out to be unnecessary, just adding a few `base_str()` were enough. I've fallen back to using the default serializer's `::write_key_val` method.

Reply:
I guess that can theoretically change the active textdomain… though perhaps the chance of getting strings with a _different_ textdomain in this contexxt is basically nil?

Reply:
seems like it.

Reply:
???

That comment is a little vague, so I'm not completely sure what you're agreeing with…

Reply:
I mean that there is a chance of strings from other textdomains to creep
in, though my tests showed nothing like that so far.